crates/sui-types/src/transaction.rs

```diff
@@ -113,6 +122,47 @@ pub enum ObjectArg {
     Receiving(ObjectRef),
 }
 
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub enum Reservation {
+    // Reserve a specific amount of the balance.
+    MaxAmount(u64),
```

Comment:
Do we want this to be `u256`, or will we add a different variant here for larger types?